dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/ShutdownServiceImpl.java

```diff
@@ -0,0 +1,192 @@
+            } catch (final Exception e) {
+                LOG.error("Failed to save job ids", e);
+            }
+        }
+
+        if (settings != null) {
+            try {
+                SettingsStore.saveSettingsStore(settings);
+            } catch (final IOException e) {
+                LOG.error("Failed to save settings information to the local filesystem", e);
+            }
+        }
+        if (jobWorkers.getTasks().size() != 0) {
+
+            final ImmutableList<Ds3JobTask> outstandingJobs = jobWorkers.getTasks().stream().collect(GuavaCollectors.immutableList());
+
+            final ShutdownTask task = new ShutdownTask(outstandingJobs);
+
+            workers.execute(task);
+            task.setOnSucceeded(event -> finalShutdown());
+        } else {
+            finalShutdown();
+        }
+    }
+
+    private void finalShutdown() {
+        workers.shutdown();
+        jobWorkers.shutdown();
+        jobWorkers.shutdownNow();
+        LOG.info("Finished shutting down");
+        Platform.exit();
+        System.exit(0);
+    }
+
+    private class ShutdownTask extends Task {
+
+        private final ImmutableList<Ds3JobTask> outstandingJobs;
+
+        public ShutdownTask(final ImmutableList<Ds3JobTask> outstandingJobs) {
+            this.outstandingJobs = outstandingJobs;
+        }
+
+
+        @Override
+        public Object call() {
+            outstandingJobs.forEach(job -> {
+                final CountDownLatch latch = new CountDownLatch(1);
+                try {
+
+                    String jobId = "";
+                    Ds3Client ds3Client = null;
+                    if (job instanceof Ds3PutJob) {
+                        final Ds3PutJob ds3PutJob = (Ds3PutJob) job;
+                        if (ds3PutJob.getJobId() != null) {
+                            jobId = ds3PutJob.getJobId().toString();
+                            ds3Client = ds3PutJob.getDs3Client();
+                        }
+                        LOG.info("Cancelled job:{} ", ds3PutJob.getJobId());
+                    } else if (job instanceof Ds3GetJob) {
+                        final Ds3GetJob ds3GetJob = (Ds3GetJob) job;
+                        if (ds3GetJob.getJobId() != null) {
+                            jobId = ds3GetJob.getJobId().toString();
+                            ds3Client = ds3GetJob.getDs3Client();
+                        }
+                        LOG.info("Cancelled job:{} ", ds3GetJob.getJobId());
+                    } else if (job instanceof RecoverInterruptedJob) {
+                        final RecoverInterruptedJob recoverInterruptedJob = (RecoverInterruptedJob) job;
+                        jobId = recoverInterruptedJob.getUuid().toString();
+                        ds3Client = recoverInterruptedJob.getDs3Client();
+                        LOG.info("Cancelled job:{} ", recoverInterruptedJob.getUuid());
+                    }
```

Comment:
This is missing an else case.  Also it looks like the getDs3Client() call can be done first before this if else chain is called.  If it cannot, we should look to see what we can do to make it work that way.